internal/config/authentication.go

```diff
@@ -123,6 +123,14 @@ type AuthenticationSession struct {
 	TokenLifetime time.Duration `json:"tokenLifetime,omitempty" mapstructure:"token_lifetime"`
 	// StateLifetime is the lifetime duration of the state cookie.
 	StateLifetime time.Duration `json:"stateLifetime,omitempty" mapstructure:"state_lifetime"`
+	// CSRF configures CSRF provention mechanisms.
+	CSRF AuthenticationSessionCSRF `json:"csrf,omitempty" mapstructure:"csrf"`
+}
+
+// AuthenticationSessionCSRF configures cross-site request forgery prevention.
+type AuthenticationSessionCSRF struct {
+	// Key is the private key string used to authenticate csrf tokens.
+	Key string `json:"-" mapstructure:"key"`
 }
```

Comment:
wonder if we should add an IT that verifies that this key is never exposed (if configured) when curling `/meta/config` ?

Reply:
That is a good idea. To avoid this regressing.

Reply:
I updated the test to both look for the key that was configured in the meta config response and ensure that the CSRF `Set-Cookie` header was supplied appropriately.